formspree/forms/views.py

```diff
@@ -366,3 +366,48 @@ def form_submissions(hashid, format=None):
                                 % (hashid, datetime.datetime.now().isoformat().split('.')[0])
                 }
             )
+
+@login_required
+def form_deletion(hashid):
+    form = Form.get_with_hashid(hashid)
+
+    if form.owner_id != current_user.id:
+        return render_template('error.html',
+                              title='Wrong user',
+                              text='You aren\'t the owner of that form.<br />Please log in as the form owner and try again.'), 400
+    if not form:
+            return render_template('error.html',
+                                   title='Not a valid form',
+                                   text='That form does not exist.<br />Please check the link and try again.'), 400
+    else:
+        for submission in form.submissions:
+            DB.session.delete(submission)
+        DB.session.delete(form)
+        DB.session.commit()
+        flash('Form successfully deleted', 'success')
+        return redirect(url_for('dashboard'))
+
+@login_required
+def submission_deletion(hashid, submissionid):
+    submission = Submission.get_by_submissionid(submissionid)
```

Comment:
The `get_by_submissionid` method is kinda useless boilerplate (I don't know if these words are offensive, but I didn't meant them to be).

The exact same behavior here can be achieved with

```python
submission = Submission.query.get(submissionid)
```